beacon_node/src/config.rs

```diff
@@ -111,10 +111,25 @@ pub fn get_config<E: EthSpec>(
     }
 
     if let Some(boot_enr_str) = cli_args.value_of("boot-nodes") {
-        client_config.network.boot_nodes = boot_enr_str
-            .split(',')
-            .map(|enr| enr.parse().map_err(|_| format!("Invalid ENR: {}", enr)))
-            .collect::<Result<Vec<Enr>, _>>()?;
+        let mut enrs: Vec<Enr> = vec![];
+        let mut multiaddrs: Vec<Multiaddr> = vec![];
+        for addr in boot_enr_str.split(',') {
+            match addr.parse() {
+                Ok(enr) => enrs.push(enr),
+                Err(_) => {
+                    // parsing as ENR failed, try as Multiaddr
+                    let multi: Multiaddr = addr
+                        .parse()
+                        .map_err(|_| format!("Not valid as ENR nor Multiaddr: {}", addr))?;
+                    if !multi.iter().any(|proto| matches!(proto, Protocol::Udp(_))) {
```

Comment:
The multiaddr here must also contain the P2p protocol for it to work as a bootnode. And when dialing it (if it has TCP, we need to strip the P2p protocol off (unless libp2p have changed in the recent versions))

Reply:
I'm using a closure that was already in the code to do all the strip + log + dial so that part should be fine